### PR TITLE
PLF-8235:500 Error message when displaying search results

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**
If we modify the space name, we search the space by the groupId

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java [hunk 1]
```diff
@@ -274,7 +274,11 @@ public Response getSpaceInfo(@Context UriInfo uriInfo,
     portalContainerName = portalName;
     
     Space space = getSpaceService().getSpaceByPrettyName(spaceName);
-    
+
+    if (space == null) {
+      space = getSpaceService().getSpaceByGroupId("/spaces/" + spaceName);
```
Comment (hbenali) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java]: String groupId = String.format("%s/%s", SpaceUtils.SPACE_GROUP, spacName);
      space = getSpaceService().getSpaceByGroupId(groupId);

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java [hunk 2]
```diff
@@ -274,7 +275,12 @@ public Response getSpaceInfo(@Context UriInfo uriInfo,
     portalContainerName = portalName;
     
     Space space = getSpaceService().getSpaceByPrettyName(spaceName);
-    
+
+    if (space == null && StringUtils.isNoneBlank(spaceName)) {
+      String groupId = String.format("%s/%s", SpaceUtils.SPACE_GROUP, spaceName);
```
Comment (hbenali) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java]: IMO, it is better to use String concatenation instead. WDYT ?

**Comment by Rdidmohamedelhadi** [comment]
If we modify the space name, we search the space by the groupId

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java [hunk 3]
```diff
@@ -274,7 +275,11 @@ public Response getSpaceInfo(@Context UriInfo uriInfo,
     portalContainerName = portalName;
     
     Space space = getSpaceService().getSpaceByPrettyName(spaceName);
-    
+
+    if (space == null && StringUtils.isNoneBlank(spaceName)) {
```
Comment (boubaker) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java]: plz avoid using `StringUtils.isNoneBlank`, use `StringUtils.isNotBlank` instead

Comment (Rdidmohamedelhadi) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java]: ok

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java [hunk 4]
```diff
@@ -274,7 +275,11 @@ public Response getSpaceInfo(@Context UriInfo uriInfo,
...
+      space = getSpaceService().getSpaceByGroupId(SpaceUtils.SPACE_GROUP + "/" + spaceName);
+    }
+
```
Comment (vsellier) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java]: Shouldn't we also test here if space is zero in case a wrong name is used. ?

Comment (Rdidmohamedelhadi) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java]: yes, i try to do it.

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java [hunk 5]
```diff
@@ -274,7 +278,15 @@ public Response getSpaceInfo(@Context UriInfo uriInfo,
...
     
     Space space = getSpaceService().getSpaceByPrettyName(spaceName);
-    
+
+    if (space == null && StringUtils.isNotBlank(spaceName)) {
+      space = getSpaceService().getSpaceByGroupId(SpaceUtils.SPACE_GROUP + "/" + spaceName);
+    }
+
+    if (space == null) {
+      return Response.status(Response.Status.NOT_FOUND).build();
```
Comment (hbenali) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java]: IMO, we shouldn't send to the client 404 Error which means the URL is broken. It is better to set 403 - Forbidden. WDYT ?

Comment (hbenali) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpacesRestService.java]: OK As we discussed with @Rdidmohamedelhadi. It is better to keep this change (404 Not Found) for security purposes

File: component/service/src/test/java/org/exoplatform/social/service/rest/SpaceRestServiceTest.java [hunk 6]
```diff
@@ -164,6 +164,46 @@ public void testGetSpacesLastVisitedAppIdNull() throws Exception {
...
+    space.setMembers(members);
+    space.setUrl("/space/" + space.getPrettyName());
+    spaceService.saveSpace(space, true);
+    space = spaceService.getSpaceByPrettyName(space.getPrettyName());
+    space.setDisplayName("test1");
+    space.setPrettyName("test1");
+    spaceService.saveSpace(space, false);
+    // Test Rest URL at org.exoplatform.social.service.rest.SpacesRestService.getSpaceInfo
+
+    String portalContainerName = "/private";
```
Comment (ahamdi) [inline component/service/src/test/java/org/exoplatform/social/service/rest/SpaceRestServiceTest.java]: Is this needed ?

File: component/service/src/test/java/org/exoplatform/social/service/rest/SpaceRestServiceTest.java [hunk 7]
```diff
@@ -164,6 +164,46 @@ public void testGetSpacesLastVisitedAppIdNull() throws Exception {
...
+
+    ContainerResponse response = service("GET", "/portal/social/spaces/spaceInfo/?spaceName=testspace", "", null, null);
```
Comment (ahamdi) [inline component/service/src/test/java/org/exoplatform/social/service/rest/SpaceRestServiceTest.java]: Remove the extra interline please.

File: component/service/src/test/java/org/exoplatform/social/service/rest/SpaceRestServiceTest.java (hunk 6 above)
Comment (Rdidmohamedelhadi) [reply component/service/src/test/java/org/exoplatform/social/service/rest/SpaceRestServiceTest.java]: I forgot it

File: component/service/src/test/java/org/exoplatform/social/service/rest/SpaceRestServiceTest.java (hunk 7 above)
Comment (Rdidmohamedelhadi) [reply component/service/src/test/java/org/exoplatform/social/service/rest/SpaceRestServiceTest.java]: ok
